File: RDataFrames/writing_root_files.cpp
#include <ROOT/RDataFrame.hxx>
#include <iostream>
#include <string>
#include <vector>

void writing_root_files()
{
  // declaring a variable for the dataframe
  ROOT::RDataFrame df("TreeS", "../Data/higgs_small.root");

  // Here we read a file and saving some of its branches to another file
  std::vector<std::string> columns = {"lepton_pT", "lepton_eta"};

  // Snapshot saves in a new root file in a new Tree
  df.Snapshot("myTree", "my_output.root", columns);
  // or we could use
  // df.Snapshot("myTree", "my_output.root", {"lepton_pT", "lepton_eta"});
}

int main()
{
  writing_root_files();

  return 0;
}

File: RDataFrames/writing_root_files.py
# Let's read the same file and tree as before
# but now just pick two branches and save the
# in a different file.

import ROOT
from ROOT import RDataFrame
df = RDataFrame('TreeS','../Data/higgs_small.root')

# Sometimes when choosing the branch name 
# python string dont work, so we are using
# a ROOT vector of string, with the desired
# branches.
columns = ROOT.vector('string')() 
columns.push_back('lepton_pT')
columns.push_back('lepton_eta')
# Snapshot saves in a new root file in a new Tree 
df.Snapshot('myTree','my_output.root',columns)
